src/Packagist/WebBundle/Package/SymlinkDumper.php

```diff
@@ -467,7 +475,17 @@ private function dumpRootFile($file)
         if (file_exists($file)) {
             rename($file, $file.'-'.time());
         }
-        $this->writeFile($file, json_encode($this->rootFile));
+
+        $json = json_encode($this->rootFile);
+        $time = time();
+        $this->writeFile($file, $json, $time);
+
+        $pathgz = $file . '.gz';
+
+        if ($this->compress && !file_exists($pathgz)) {
```

Comment:
I'd remove the pathgz. We should always recompress if we rewrite the file. Avoids a lot of potential fuckups.